v0/destinations/intercom/transform.js

```diff
@@ -1,250 +1,379 @@
-  return rawPayload;
+  throw new Error("Email or userId is mandatory");
 }
 
-function getTrackPayload(message, intercomConfig) {
-  const rawPayload = {};
-  const properties = get(message, "properties")
-    ? Object.keys(message.properties)
-    : undefined;
-
-  if (properties) {
-    const metadata = {};
-    const price = {};
-    const order_number = {};
-
-    properties.forEach(property => {
-      const value = message.properties[property];
-
-      metadata[property] = value;
-      if (property === "price" || property === "currency") {
-        updatePayload(property, mapPayload.track.price, value, price);
-        price.amount *= 100;
-        metadata.price = price;
-      }
-
-      if (property === "order_ID" || property === "order_url") {
-        updatePayload(property, mapPayload.track.order, value, order_number);
-        metadata.order_number = order_number;
-      }
-    });
-    rawPayload.metadata = metadata;
+function validateAndBuildResponse(message, payload, category, destination) {
+  const messageType = message.type.toLowerCase();
+  const response = defaultRequestConfig();
+  switch (messageType) {
+    case EventType.IDENTIFY:
+      response.body.JSON = removeUndefinedAndNullValues(
+        validateIdentify(message, payload)
+      );
+      break;
+    case EventType.TRACK:
+      response.body.JSON = removeUndefinedAndNullValues(
+        validateTrack(message, payload)
+      );
+      break;
+    default:
+      throw new Error("Message type not supported");
   }
 
-  rawPayload.event_name = message.event ? message.event : undefined;
-  rawPayload.user_id = message.userId ? message.userId : message.anonymousId;
-  rawPayload.created_at = Math.floor(
-    new Date(message.originalTimestamp).getTime() / 1000
-  );
-
-  return rawPayload;
+  response.method = defaultPostRequestConfig.requestMethod;
+  response.endpoint = category.endpoint;
+  response.headers = {
+    "Content-Type": "application/json",
+    Authorization: `Bearer ${destination.Config.accessToken}`,
+    Accept: "application/json"
+  };
+  response.userId = message.anonymousId;
+  return response;
 }
 
-function getTransformedJSON(message, intercomConfig) {
-  let rawPayload;
-  switch (message.type) {
-    case EventType.TRACK:
-      rawPayload = getTrackPayload(message, intercomConfig);
-      break;
+function processSingleMessage(message, destination) {
+  if (!message.type) {
+    throw Error("Message Type is not present. Aborting message.");
+  }
+  const messageType = message.type.toLowerCase();
+  let category;
+
+  switch (messageType) {
     case EventType.IDENTIFY:
-      rawPayload = getIdentifyPayload(message, intercomConfig);
+      category = ConfigCategory.IDENTIFY;
+      break;
+    case EventType.TRACK:
+      category = ConfigCategory.TRACK;
       break;
     case EventType.GROUP:
-      rawPayload = getGroupPayload(message, intercomConfig);
+      category = ConfigCategory.GROUP;
```

Comment:
but group calls are not dealt with later. so shouldn't we work on it or reject it here only?